tiledb/test-utils/src/signal.rs

```diff
@@ -0,0 +1,232 @@
+use std::pin::Pin;
+use std::sync::Mutex;
+
+pub use nix::sys::signal::{SigHandler, Signal};
+
+/// Overrides a signal handler within its scope.
+pub struct SignalGuard {
+    signo: Signal,
+    restore_handler: SigHandler,
+}
+
+impl SignalGuard {
+    pub fn new(signo: Signal, handler: SigHandler) -> Self {
+        let restore_handler =
+            unsafe { nix::sys::signal::signal(signo, handler) }
+                .expect("Error installing signal handler");
+
+        SignalGuard {
+            signo,
+            restore_handler,
+        }
+    }
+}
+
+impl Drop for SignalGuard {
+    fn drop(&mut self) {
+        unsafe {
+            nix::sys::signal::signal(self.signo, self.restore_handler)
+                .expect("Error restoring previous signal handler");
+        }
+    }
+}
+
+struct RawSignalCallback(*mut ());
+
+impl<'a> From<&mut SignalCallback<'a>> for RawSignalCallback {
+    fn from(value: &mut SignalCallback<'a>) -> Self {
+        let ptr = value as *const SignalCallback<'a>;
+        RawSignalCallback(unsafe { std::mem::transmute::<_, _>(ptr) })
+    }
+}
+
+unsafe impl Send for RawSignalCallback {}
+
+const SIGNAL_CALLBACK_DEFAULT: Mutex<Option<RawSignalCallback>> =
+    Mutex::new(None);
+static SIGNAL_CALLBACKS: [Mutex<Option<RawSignalCallback>>; 32] =
+    [SIGNAL_CALLBACK_DEFAULT; 32];
```

Comment:
Is this documented as a constant limit across all of the various *nix platforms? It appears to be true on Darwin, but I'm not sure if there's a BSD out there with more than 32 signals.

Reply:
Probably not, but it is a limit inside the `nix` crate.
https://docs.rs/nix/latest/nix/sys/signal/enum.Signal.html